src/Mod/Sketcher/Gui/DrawSketchDefaultWidgetHandler.h

```diff
@@ -644,16 +644,7 @@ class DrawSketchDefaultWidgetHandler: public DrawSketchDefaultHandler<HandlerT,
          */
         void onHandlerModeChanged()
         {
-            if constexpr (std::is_same_v<StateMachines::OneSeekEnd, SelectMode>) {

```

Comment:
Because Point tool uses OneSeekEnd and Point tool does require a behaviour different than default, here the choice was to remove the default behaviour, so as not to explicitly provide a behaviour different than default for point.

If another tool with OneSeekEnd is created, and uses the default behaviour, it will have to implement it, as it is not there anymore. This is inconsistent with other state machines.

There are two approaches: 
(i) we do not provide default behaviours at all and each handler will have to implement its own.
(ii) we provide consistent default behaviours for several state machines.

If we take (ii) is because we assume most tools would use a default behaviour, and it also helps as documentation of how the function should normally work. However, if we take (ii) the default behaviour needs to be consistent for all state machines and any tool using different shall provide its own.

If we take (i) is because there is not a "logical default pattern".

Reply:
I think that actually it is better to do (i) and not provide defaults.
While it sounds like a good idea, in my experience it is not ideal:
- First, having writen most of the DSH before, there was actually very little use of the default if I recall correctly. We have not so many DSH, so in the end there wasn't two really similar. If we had 100 DSH it would make sense I guess.
- It increase complexity, making debugging harder. For the offset tool for example I had a bug that took me a long time to identify (which is the OneSeekEnd default I removed).

So yes I would rather remove the defaults.

Reply:
I am ok with that. But then there is a second step, where we have to decide from which functions we remove the defaults. It may be from all functions, or it may be from some.

I will be posting a working branch, as I have finished with the toolwidget signaling block. Then we can discuss over it.

Reply:
I think we can remove them most for now. And if need be we can reimplement the one that make sense.

We can already have a look between the point DSH and the offset DSH, which are IIRC the only 2 tools which are OneSeekEnd. And see between these two tools for what function we use the default. I would guess that its stuff related to the construction mode combo box. Then we can keep these.